proposed/2022/version-overrides.md

```diff
@@ -0,0 +1,119 @@
+# Title
+
+- [Jeff Kluge](https://github.com/jeffkl) & [Jon Douglas](https://github.com/jondouglas)
+- Start Date (2022-02-07)
+- [11516](https://github.com/NuGet/Home/issues/11516)
+- [4426](https://github.com/NuGet/NuGet.Client/pull/4426)
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+When specifying package versions in a central location, developers want the to be able to override versions for a particular project.  This is useful when a project is not directly tied to the rest of the repo and needs to use a newer version of a package.  Additionally, some projects have a need to specify an older version of a package when targeting an older version of .NET Framework.
+
+This proposal introduces the ability to override a centrally defined package version on a `<PackageReference />` item in a well documented and supported way.
+
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+When interviewing a number of developers who currently use the NuGet provided central package management feature, we found that the majority of them were frustrated with the lack of a way to override a centrally defined package version for a particular project. Many of these individuals would use the MSBuildSdks [Microsoft.Build.CentralPackageVersions](https://github.com/microsoft/msbuildsdks) as an alternative to give them the flexibility they needed in their repositories.
+
+The concept of dependency resolution overrides exists in many ecosystems and for NuGet, we want to ensure that there is a formal way to override package versions in the case that:
+
+- A project exists in a repository but is not part of the main set of shipping assemblies and is allowed to use newer, untested package depedencies (such as an experimental tool)
+- A project that has more advanced requirements and cannot use a single version of a package dependency for all of its target frameworks
+
+In these cases, a project can override the centrally defined package version.  However, if that project is part of a transitive graph with version conflicts, a user would still need to resolve those conflicts before restore would succeed.  The main point of this feature is to give a well documented and supported way of overriding a package version for a particular project, not to make graph resolution easier if package conflicts exist.  
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+First a user opts-in to Central Package Management with a `Directory.Packages.props` file in their directory tree.
```

Comment:
Given that `VersionOverride` only works when CPM is enabled, should we warning/error if `VersionOverride` is used when CPM is not used? 

I'm thinking specifically of a scenario where someone new to .NET copy pastes lines from existing csproj files into a new project they created themselves, without understanding NuGet's features. If they did that with this proposal, they will get a very obscure warning/error message which will not be clear at all what the root cause is.

Reply:
I agree with this. 

Being in different modes has been something people that were trying to migrate to the originjal version of CPVM hit a lot.

Reply:
I'll put a note in the spec to emit a warning when `VersionOverride` is ignored du to CPM being disabled.